src/webgpu/api/operation/command_buffer/image_copy.spec.ts

```diff
@@ -627,6 +629,250 @@ class ImageCopyTest extends GPUTest {
+            fn main() -> [[location(0)]] vec4<f32> {
+              return textureLoad(storageImage, vec2<i32>(param.stencilBitIndex, 0));
+            }`,
+        }),
+        entryPoint: 'main',
+        targets: [
+          {
+            // As we implement "rendering one bit in each draw() call" with blending operation
+            // 'add', the format of outputTexture must support blending.
+            format: 'r8unorm',
+            blend: {
+              color: { srcFactor: 'one', dstFactor: 'one', operation: 'add' },
+              alpha: { srcFactor: 'one', dstFactor: 'one', operation: 'add' },
+            },
+          },
+        ],
+      },
+
+      primitive: {
+        topology: 'triangle-list',
+      },
+    };
+
+    // Prepare the bindGroup that contains uniformBuffer and referenceTexture.
+    const bindGroup = this.device.createBindGroup({
+      layout: bindGroupLayout,
+      entries: [
+        {
+          binding: 0,
+          resource: {
+            buffer: uniformBuffer,
+            size: 4,
+          },
+        },
+        {
+          binding: 1,
+          resource: referenceTexture.createView(),
+        },
+      ],
+    });
+
+    // "Copy" the stencil value into the color attachment with 8 draws in one render pass. Each draw
+    // will "Copy" one bit of the stencil value into the color attachment. The bit of the stencil
+    // value is specified by setStencilReference().
+    const outputTextureSize = [copyLayout.mipSize[0], copyLayout.mipSize[1], 1];
+    const outputTexture = this.device.createTexture({
+      format: 'r8unorm',
+      size: outputTextureSize,
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+
+    for (
+      let stencilTextureLayer = 0;
+      stencilTextureLayer < stencilTextureSize[2];
+      ++stencilTextureLayer
+    ) {
+      const encoder = this.device.createCommandEncoder();
+      const renderPass = encoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: outputTexture.createView(),
+            loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 0.0 },
+            storeOp: 'store',
+          },
+        ],
+        depthStencilAttachment: {
+          view: stencilTexture.createView({
+            baseMipLevel: stencilTextureMipLevel,
+            mipLevelCount: 1,
+            baseArrayLayer: stencilTextureLayer,
+            arrayLayerCount: 1,
+          }),
+          stencilLoadValue: 'load',
+          stencilStoreOp: 'store',
+          depthLoadValue: 0,
+          depthStoreOp: 'store',
+        },
+      });
+
+      const depthStencilStateBase: GPUDepthStencilState = {
+        format: stencilTextureFormat,
+        stencilFront: {
+          compare: 'equal',
+        },
+        stencilBack: {
+          compare: 'equal',
+        },
+      };
+      for (let stencilBitIndex = 0; stencilBitIndex < stencilBitCount; ++stencilBitIndex) {
+        const depthStencilState = depthStencilStateBase;
```

Comment:
u-nit: I think it might be more idiomatic JS to first create the full descriptor and continuously update it here. Especially since the rest of the GPUDepthStencilState doesn't change:

```
renderPipelineDescriptor.depthStencil.stencilReadMask = 1 << stencilBitIndex;
const renderPipeline = this.device.createRenderPipeline(renderPipelineDescriptor);
```